init.fish

```diff
@@ -0,0 +1,8 @@
+function zoxide-add --on-event fish_prompt
```

Comment:
What is the benefit of using the `fish_prompt` event, as opposed to watching the `PWD` shell variable?

Ref: https://github.com/jethrokuan/z/blob/5e8d0e2c900d6a8c9704e153a8126f776d9d2f3c/conf.d/z.fish#L42

Reply:
I don't think there's any benefit, It's just what makes sense to me. I want folders to be added to zoxide when I `cd` voluntarily, not when the `$PWD` is changed which can happen multiple times in a function without the prompt being updated.

Reply:
I've never used any functions that used `cd` in their internals, so I failed to take that into account. Fair enough. Thanks for your work on this!

Reply:
It's not common for functions to do that, but it can happen so I thought I should make sure the zoxide database doesn't get unwanted entries. 😊